docs/proposals/kep-003-service-resouce-aggregation.md

```diff
@@ -0,0 +1,57 @@
+# [KEP-003] Service resource aggregation
+
+## Background
+
+Currently, we're creating catapult deployment per each `CustromResourceDiscovery` object. The catapult component transports custom resources from the service cluster to the management cluster, and vice versa, asynchronously. For example the `db.example.io` apiGroup in the service cluster `svc-1` becomes `db.internal.svc-1.provider-1` apiGroup in the management cluster.
+
+Furthermore, catapult component handles the webhooks for internal apiGroup. If there is `CustromResourceDiscoverySet` there's a new catapult deployment per each service cluster offering the same service.
+
+* See also:
+
+https://github.com/kubermatic/kubecarrier/issues/316
+
+## Issues
+
+The issues with this approach are multi-fold:
+
+* propagation is asynchronously, whereas management clusters serves as a "cache" to service cluster's ground truth. In more complicated setups the management cluster's resource version and service cluster resource version could diverge in unreconciable way. To give a simple, albeit toy example consider the following:
+
+    * There's deployment on the service cluster. This resource is propagated to the management cluster. At the start the deployment replicas is 1.
+       * HPA controller on the service cluster increases the replica count to 2
+       * User on the management cluster changes the deployment image
+
+   Now we have to two diverging versions what the deployment should be, and how should it look like. Somehow we have to perform 2-way, or 3-way merge between those objects adding additional complexity.
+
+   Note: In the common case the service cluster operators would only change the `.status` subresource, while the management cluster would handle the `.spec` field.
+
+* webhook complexity
+
+    We're running semi-complex code for object updating and creation webhooks. We rely on `dry-run` support from the service cluster. Service cluster resource might not support some critical `dry-run` operation, thus our webhooks won't behave as if they are being run in the service cluster in the first place.
```

Comment:
What do you mean that "resource might not support some critical `dry-run` operation"?
But yeah, you are right, the KubeCarrier dynamic webhooks require some extra conifgurations of service cluster webhook, for example, the `SideEffects` should be configured to `NoneOnDryrun`, otherwise it will reject request with dry-run flag.
ref: https://github.com/kubermatic/kubecarrier/pull/396

Reply:
Exactly that, there's some webhook without the `SideEffect: NoneOnDryRun` which would run upon creation, but it's actually mislabeled `dry-run`.